lib/events/athena/consumer.go

```diff
@@ -631,6 +636,38 @@ func (s *sqsMessagesCollector) downloadEventFromS3(ctx context.Context, payload
 	return buf.Bytes(), nil
 }
 
+// fanOutEvents takes events channel and broadcast it to out1 and our2.
+// It closes out channels when it stops receiving from in.
+func fanOutEvents(in <-chan eventAndAckID, out1, out2 chan<- eventAndAckID) {
+	for ev := range in {
+		out1 <- ev
+		out2 <- ev
+	}
+
+	close(out1)
+	close(out2)
+}
+
+// receiveMetrics is collecting events from eventsCh and publishing metrics based on it.
+func receiveMetrics(eventsCh <-chan eventAndAckID) {
+	var count, size int
+	var oldestEventTimestamp time.Time
+	for ev := range eventsCh {
+		consumerLastEventSeen.SetToCurrentTime()
+		count++
+		size += ev.event.Size()
+		if oldestEventTimestamp.IsZero() || oldestEventTimestamp.After(ev.event.GetTime()) {
+			oldestEventTimestamp = ev.event.GetTime()
+		}
+	}
+	if count == 0 {
+		return
+	}
+	consumerBatchCount.Add(float64(count))
+	consumerBatchSize.Observe(float64(size))
+	consumerAgeOfOldestProcessedMessage.Set(time.Since(oldestEventTimestamp).Seconds())
```

Comment:
nit: 
What would you say about moving metrics logic into `sqsMessagesCollector.fromSQS` instead of creating a artificial pipeline ?